layer/contracts/datasets.py

```diff
@@ -25,10 +25,10 @@ class DatasetBuildStatus(enum.IntEnum):
 
 @dataclass(frozen=True)
 class DatasetBuild:
-    id: uuid.UUID = field(default_factory=uuid.uuid4)
-    status: DatasetBuildStatus = DatasetBuildStatus.INVALID
+    id: uuid.UUID
+    status: DatasetBuildStatus
+    tag: str
```

Comment:
the same `DatasetBuild` is also used inside `Context` but actually there status does not make sense: we are gonna need to distinguish the execution context (run/experiment) from artifacts yes (tags)

Reply:
yes, absolutely. that is hopefully coming soon